src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -430,6 +430,20 @@ protected override bool FindFirstChar()
                     }
                 }
             }
+            else if (RegexCharClass.IsDoubleton(set))
```

Comment:
The IsDoubleton and IsSingleton call share some if checks. Maybe there is a way to converge both calls to reduce the number of redundant calls?

Reply:
I could pull out `set[FLAGS] == 0 && set[CATEGORYLENGTH] == 0 && !IsSubtraction(set)` and call it in 3 places. Assuming it is inlined that would be fine. I'm not sure what it would be called though - might not be worth it?

I could add similar inline comments to other related functions in there though.